packages/push-notifications/src/services/PushNotificationsService.ts

```diff
@@ -0,0 +1,45 @@
+import { ConnectionService } from '@aries-framework/core'
+import { Lifecycle, scoped } from 'tsyringe'
+
+import {
+  PushNotificationsGetDeviceInfoMessage,
+  PushNotificationsSetExpoDeviceInfoMessage,
+  PushNotificationsSetFcmDeviceInfoMessage,
+  PushNotificationsSetNativeDeviceInfoMessage,
+} from '../messages'
+
+export type DeviceVendor = 'android' | 'ios' | string
+
+export interface DeviceInfo {
+  deviceToken: string
+  deviceVendor: DeviceVendor
+}
+
+@scoped(Lifecycle.ContainerScoped)
+export class PushNotificationsService {
+  public constructor(private connectionService: ConnectionService) {}
+
+  public async createSetNativeDeviceInfo(connectionId: string, deviceInfo: DeviceInfo) {
+    const message = new PushNotificationsSetNativeDeviceInfoMessage(deviceInfo)
+    const connection = await this.connectionService.getById(connectionId)
+    return { message, connection }
+  }
```

Comment:
Sorry for not mentioning last review but we usually use ids on the module layer and records on the service layer. Then you also don't need to return the connection record, as the module already has it. Then we should probably also assert the connection is ready 

```suggestion
  public async createSetNativeDeviceInfo(connectionRecord: ConnectionRecord, deviceInfo: DeviceInfo) {
    return new PushNotificationsSetNativeDeviceInfoMessage(deviceInfo)
  }
```

e.g. this is from the discover features module:

```
  public async queryFeatures(connectionId: string, options: { query: string; comment?: string }) {
    const connection = await this.connectionService.getById(connectionId)
    
    // This is not in the discover features module, but we should check if the connection is ready
    connection.assertReady()

    const queryMessage = await this.discoverFeaturesService.createQuery(options)

    const outbound = createOutboundMessage(connection, queryMessage)
    await this.messageSender.sendMessage(outbound)
  }
```

Reply:
Updated because we don't even need the connection record in the service now anymore